builds/mariner/SPECS/aziot-edge/aziot-edge.spec

```diff
@@ -0,0 +1,207 @@
+%define iotedge_user iotedge
+%define iotedge_group %{iotedge_user}
+%define iotedge_home %{_localstatedir}/lib/aziot/edged
+%define iotedge_logdir %{_localstatedir}/log/aziot/edged
+%define iotedge_socketdir %{_localstatedir}/lib/iotedge
+%define aziot_confdir %{_sysconfdir}/aziot
+%define iotedge_confdir %{aziot_confdir}/edged
+
+Name:           aziot-edge
+Version:        @@VERSION@@
+Release:        @@RELEASE@@%{?dist}
+
+License:        MIT
+Summary:        Azure IoT Edge Module Runtime
+URL:            https://github.com/azure/iotedge
+
+%{?systemd_requires}
+BuildRequires:  systemd
+Requires(pre):  shadow-utils
+Requires:       openssl
+Requires:       aziot-identity-service
+Requires:       moby-engine
+Requires:       moby-cli
```

Comment:
docker is required for iotedge to work right? otherwise the modules would not run. I believe these are the only moby components required for that functionality.  by default, mariner does not have docker installed not sure if that is the same for other linux distros.

Reply:
if you want we could dynamically add in the azure-identity-service version, just update date it similar to how we handle the version. the version of azure-identity-service is locked in the background by the step where we download that package in the package build yaml.

Reply:
> docker is required for iotedge to work right? otherwise the modules would not run. I believe these are the only moby components required for that functionality. by default, mariner does not have docker installed not sure if that is the same for other linux distros.

We require the moby-engine & moby-cli at runtime to run docker images. However, we do not require those during the compile time for edgelet. Let's try removing these and rerun the pipeline. They *should* pass.

Reply:
> if you want we could dynamically add in the azure-identity-service version, just update date it similar to how we handle the version. the version of azure-identity-service is locked in the background by the step where we download that package in the package build yaml.

Yeah. Let's go ahead and do that. I prefer it to be explicitly called out both during the compile time and the installation time. You're correct that during the build time, we implicitly fix the version of the IIS source. However, this parameter is carried over to the output *.rpm which later will be checked when you do `rpm install`. Right now, any version of IIS can interchangeably use with aziot-edge for Mariner OS and I'm not a big fan of that.

Reply:
> We require the moby-engine & moby-cli at runtime to run docker images. However, we do not require those during the compile time for edgelet. Let's try removing these and rerun the pipeline. They _should_ pass.

That should be the behavior we are seeing here. the requires flag says that these packages were required at install time and not build time. If this was going to be installed at build time it would have to be specified as BuildRequires. requires just means that the packages need to be accessible but are not actually installed by the mariner toolkit. This ensures that the packages does not require a package that does not exists and thus can never be installed.